corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -14,14 +14,15 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
         initialPageData = hqImport("hqwebapp/js/initial_page_data");
 
     // special format handled by CaseSearch API
-    var encodeValue = function (model, value, searchForBlank) {
+    var encodeValue = function (model, searchForBlank) {
+            var value = model.get('value');
             if (value && model.get("input") === "daterange") {
                 value = "__range__" + value.replace(separator, "__");
-            } else if (model.get('input') === 'select') {
+            } else if (value && model.get('input') === 'select') {
```

Comment:
When pulled from HTML, the value would always be defined, if only as a blank string. Now that it's pulled from the model, it can be null, so we need an extra guard here.